pkg/metadata/main_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+)
+
+func TestWriteConfig(t *testing.T) {
+	basePath, err := ioutil.TempDir(os.TempDir(), "metadata")
+	if err != nil {
+		t.Fatalf("can't make a temp rootdir %v", err)
+	}
+	defer os.RemoveAll(basePath)
+
+	json := `{"ssh":{"sshd_config":{"perm":"0600","content":"PermitRootLogin yes\nPasswordAuthentication no"}},"foo":{"bar":"foobar","baz":{"perm":"0600","content":"bar"}}}`
```

Comment:
AIUI using backticks to quote in go gives you the opportunity to format things a bit more readable, e.g. to use raw newlines and indentation.